docs/src/pages/components/radio-buttons/CustomizedRadios.js

```diff
@@ -23,13 +23,18 @@ const BpIcon = styled('span')(({ theme }) => ({
     outline: '2px auto rgba(19,124,189,.6)',
     outlineOffset: 2,
   },
-  'input:hover ~ &': {
-    backgroundColor: theme.palette.mode === 'dark' ? '#30404d' : '#ebf1f5',
+  '&:hover': {
+    'span:empty:not([class*=MuiTouchRipple-root])': {
+      zIndex: 0,
+      backgroundColor: theme.palette.mode === 'dark' ? '#30404d' : '#ebf1f5',
+    },
   },
-  'input:disabled ~ &': {
-    boxShadow: 'none',
-    background:
-      theme.palette.mode === 'dark' ? 'rgba(57,75,89,.5)' : 'rgba(206,217,224,.5)',
+  '&.Mui-disabled': {
+    'span:empty:not([class*=MuiTouchRipple-root])': {
+      boxShadow: 'none',
+      backgroundColor:
+        theme.palette.mode === 'dark' ? 'rgba(57,75,89,.5)' : 'rgba(206,217,224,.5)',
+    },
```

Comment:
Please revert all these CSS changes, the objective is to simplify the DX

Reply:
The objective as stated in the issue was to make it so that the checkboxes and radio buttons were not clickable through overlays, while keeping them functioning the same otherwise. If that wasn't the objective, sorry I misunderstood.

Reply:
That being said, I know these css changes complicated things a bit - I'd been going off of your recommendation to move where the checked item lived in the SwitchBase component - problematically it moved items around such that the css wasn't targeting them correctly anymore, hence the changes. I'll see if I can come up with a better solution!